### PR TITLE
fix(better-define): cannot import type from vue file

### LINKED ISSUE #208: Can't import from vue file (Nuxt)
### Describe the bug

When I import interface from vue file, I got error:
```
warning: unplugin-vue-better-define SyntaxError: Unexpected token, expected "}" (3:7)
  Plugin: unplugin-vue-better-define
  File: /home/f3oall/coding/js/nuxt/examples/a.vue
                                                                      11:34:33

 WARN  Unexpected token, expected "}" (3:7)                           11:34:35

  at instantiate (node_modules/@babel/parser/lib/index.js:67:32)
  at constructor (node_modules/@babel/parser/lib/index.js:364:12)
  at JSXParserMixin.raise (node_modules/@babel/parser/lib/index.js:3363:19)
  at JSXParserMixin.unexpected (node_modules/@babel/parser/lib/index.js:3396:16)
  at JSXParserMixin.expect (node_modules/@babel/parser/lib/index.js:3758:28)
  at JSXParserMixin.jsxParseExpressionContainer (node_modules/@babel/parser/lib/index.js:7095:10)
  at JSXParserMixin.jsxParseElementAt (node_modules/@babel/parser/lib/index.js:7173:36)
  at JSXParserMixin.jsxParseElement (node_modules/@babel/parser/lib/index.js:7218:17)
  at JSXParserMixin.parseExprAtom (node_modules/@babel/parser/lib/index.js:7231:19)
  at JSXParserMixin.parseExprSubscripts (node_modules/@babel/parser/lib/index.js:11182:23)
```

Import from .ts works as expected. If it's not a bug, I suggest extending the docs, because it's not clear that you can't import from .vue files.

### Reproduction

https://github.com/f3oall/vue-macto-reproduction

### System Info

```Shell
System:
    OS: Linux 5.15 Ubuntu 22.04.1 LTS 22.04.1 LTS (Jammy Jellyfish)
    CPU: (12) x64 12th Gen Intel(R) Core(TM) i7-1255U
    Memory: 3.99 GB / 7.62 GB
    Container: Yes
    Shell: 5.8.1 - /usr/bin/zsh
  Binaries:
    Node: 19.2.0 - ~/.nvm/versions/node/v19.2.0/bin/node
    npm: 8.19.3 - ~/.nvm/versions/node/v19.2.0/bin/npm
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/sxzz/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/sxzz/contribute).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

### PR DESCRIPTION
**Pull request by alexzhang1030**
### Description

Extract `setupScript` content if they import type from `vue` file.

### Linked Issues

closes #208

### CODE REVIEW

File: packages/better-define/tests/fixtures/export-type.exclude.vue [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<script setup lang="ts">
```
Comment (sxzz) [inline packages/better-define/tests/fixtures/export-type.exclude.vue:1]: We should also support normal script tag `<script>`

File: packages/common/src/vue.ts [hunk 2]
```diff
@@ -45,6 +47,28 @@ export function parseSFC(code: string, id: string): SFC {
   }
 }
 
+/**
+ * @param code origin
+ * @param id fileName
+ * @returns {{content: string, lang: string | undefined}}
+ */
+export function getSetupScriptContentAndLang(
```
Comment (sxzz) [inline packages/common/src/vue.ts:55]:
```suggestion
export function getFileCodeAndLang(
```

File: packages/common/src/vue.ts [hunk 3]
```diff
@@ -45,6 +47,28 @@ export function parseSFC(code: string, id: string): SFC {
   }
 }
 
+/**
+ * @param code origin
+ * @param id fileName
+ * @returns {{content: string, lang: string | undefined}}
+ */
```
Comment (sxzz) [inline packages/common/src/vue.ts:54]: This part is unnecessary.

File: packages/common/src/vue.ts [hunk 4]
```diff
@@ -45,6 +47,28 @@ export function parseSFC(code: string, id: string): SFC {
...
+export function getSetupScriptContentAndLang(
+  code: string,
+  id: string
+): { content: string; lang?: string } {
```
Comment (sxzz) [inline packages/common/src/vue.ts:58]:
```suggestion
): { content: string; lang: string } {
```

defaults to `js`

**Comment by alexzhang1030** [comment]
Perhaps need improve docs , tell users what type of typing defininitions can handle.

File: packages/common/src/vue.ts [hunk 5]
```diff
@@ -47,25 +47,17 @@ export function parseSFC(code: string, id: string): SFC {
...
-    return {
-      content: sfc.scriptSetup.content ?? code,
-      lang: sfc.lang,
-    }
   return {
-    content: code,
-    lang: sfc.lang,
+    content: sfc.scriptSetup
+      ? sfc.scriptSetup.content
+      : sfc.script?.content ?? '',
```
Comment (sxzz) [inline packages/common/src/vue.ts]: I mean we should support both of them. Concat their's content.

Comment (alexzhang1030) [reply packages/common/src/vue.ts]: Great 👍 ! I forget about it.

File: packages/common/src/vue.ts [hunk 6]
```diff
@@ -53,8 +53,11 @@ export function getFileCodeAndLang(
 ): { code: string; lang: string } {
   if (!REGEX_VUE_SFC.test(id)) return { code, lang: getLang(id) }
   const sfc = parseSFC(code, id)
+  const scriptCode = sfc.script?.content ?? ''
   return {
-    code: sfc.scriptSetup ? sfc.scriptSetup.content : sfc.script?.content ?? '',
+    code: sfc.scriptSetup
+      ? `${scriptCode}\n${sfc.scriptSetup.content}`
```
Comment (sxzz) [inline packages/common/src/vue.ts:59]:
```suggestion
      ? `${scriptCode}\n;\n${sfc.scriptSetup.content}`
```

**Review by sxzz — APPROVED** [review]

**Pull request merged**